invoice.py

```diff
@@ -222,23 +243,43 @@ def generate_facturae_wizard(cls, invoices):
         pass
 
     @classmethod
-    def generate_facturae_default(cls, invoices, certificate_password):
+    def generate_facturae(cls, invoices, certificate=None, service=None):
+        Configuration = Pool().get('account.configuration')
+
+        config = Configuration(1)
+        transaction = Transaction()
+        context = transaction.context
+
         to_write = ([],)
         for invoice in invoices:
             if invoice.invoice_facturae:
                 continue
             facturae_content = invoice.get_facturae()
             invoice._validate_facturae(facturae_content)
             if backend.name != 'sqlite':
-                invoice_facturae = invoice._sign_facturae(
-                    facturae_content, certificate_password)
+                invoice_facturae = invoice._sign_facturae(facturae_content,
+                    'default', certificate)
             else:
                 invoice_facturae = facturae_content
             to_write[0].append(invoice)
             to_write += ({'invoice_facturae': invoice_facturae},)
         if to_write[0]:
             cls.write(*to_write)
 
+        # send facturae to service
+        if not service and config.facturae_service:
+            service = config.facturae_service
+        if service:
+            with transaction.set_context(
+                    # queue_scheduled_at=config.invoice_facturae_after,
```

Comment:
AFAIU we should use the delay here

Reply:
322fca81de602b0681b7d8b7312e27cd0e260ec0